### ISSUE TITLE
httpd pid directory always deleted on vagrant halt on Centos 7.2

### ISSUE THREAD

**Issue by odoell** [chef-cookbooks/httpd#66]
Hello,

i install and start the httpd service with following cookbook recipe:

```ruby
# Install Apache and start the service.
httpd_service 'default' do
  mpm 'prefork'
  action [:create, :start]
end

# Install the mod_php Apache module.
httpd_module 'php' do
  instance 'default'
end

# Install gd.
package 'gd' do
  action :install
end

# Install gd-devel.
package 'gd-devel' do
  action :install
end

# Install php-gd.
package 'php-gd' do
  action :install
end

# Install php-mysql.
package 'php-mysql' do
  action :install
  notifies :restart, 'httpd_service[default]'
end

# Create vhost directories
if node["streaming_evo"]["sites"]

  node["streaming_evo"]["sites"].each do |index, site|

    htdocs = defined?(site["vhost"]["document_root"]) ? site["vhost"]["document_root"] : index

    # Avoid potential duplicate slash in docroot path from config.json input.
    if htdocs.start_with?("/")
      htdocs = htdocs[1..-1]
    end

    # Create subidrectores, allow for multiple layers deep.
    htdocs = "var/www/" + htdocs + "/public_html"
    htdocs = htdocs.split(%r{\/\s*})
    folder = "/"
    for i in (0..htdocs.length - 1)
      folder = folder + htdocs[i] + "/"
      directory folder do
        owner node['streaming_evo']['user']
        group node['streaming_evo']['group']
        mode "0755"
        action :create
      end
    end
  end
end

# Write the home page.
template "#{node['streaming_evo']['document_root']}/index.php" do
  source 'index.php.erb'
  mode '0644'
  owner node['streaming_evo']['user']
  group node['streaming_evo']['group']
  variables(
    :database_password => node['streaming_evo']['mysql_admin_password']
  )
end

# Add the site configuration.
httpd_config 'default' do
  instance 'default'
  source 'vhost.conf.erb'
  notifies :restart, 'httpd_service[default]'
end
```

After a first vagrant.up and provisioning all works as intended, the local box is reachable via IP or domain names. That's great.

But after a vagrant.halt and vagrant.up the /var/run/httpd-default directory is missing and the httpd-default.pid file cannot be written so starting httpd-default.service quits with an error.

When i do a vagrant.provision on the box again, the directory is created and the httpd-default.pi file can be written.

This gives a systemctl status httpd-default.service
```
[vagrant@default-centos-72 ~]$ systemctl status httpd-default.service
â— httpd-default.service - The Apache HTTP Server
   Loaded: loaded (/usr/lib/systemd/system/httpd-default.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Thu 2016-02-25 13:12:22 UTC; 47s ago
  Process: 1020 ExecStop=/bin/kill -WINCH ${MAINPID} (code=exited, status=1/FAILURE)
  Process: 801 ExecStart=/usr/sbin/httpd-default -f /etc/httpd-default/conf/httpd.conf -DFOREGROUND (code=exited, status=1/FAILURE)
 Main PID: 801 (code=exited, status=1/FAILURE)
```

And the content of /var/log/httpd-default/error_log

```
[Thu Feb 25 12:26:41.747041 2016] [mpm_prefork:notice] [pid 5250] AH00163: Apache/2.4.6 (CentOS) configured -- resuming normal operations
[Thu Feb 25 12:26:41.747095 2016] [core:notice] [pid 5250] AH00094: Command line: '/usr/sbin/httpd-default -f /etc/httpd-default/conf/httpd.conf -D FOREGROUND'
[Thu Feb 25 12:26:41.881516 2016] [mpm_prefork:notice] [pid 5250] AH00171: Graceful restart requested, doing restart
[Thu Feb 25 12:26:41.884327 2016] [mpm_prefork:notice] [pid 5250] AH00163: Apache/2.4.6 (CentOS) configured -- resuming normal operations
[Thu Feb 25 12:26:41.884339 2016] [core:notice] [pid 5250] AH00094: Command line: '/usr/sbin/httpd-default -f /etc/httpd-default/conf/httpd.conf -D FOREGROUND'
[Thu Feb 25 12:28:04.732534 2016] [mpm_prefork:notice] [pid 5250] AH00170: caught SIGWINCH, shutting down gracefully
[Thu Feb 25 12:28:05.777240 2016] [mpm_prefork:notice] [pid 6742] AH00163: Apache/2.4.6 (CentOS) PHP/5.4.16 configured -- resuming normal operations
[Thu Feb 25 12:28:05.777269 2016] [core:notice] [pid 6742] AH00094: Command line: '/usr/sbin/httpd-default -f /etc/httpd-default/conf/httpd.conf -D FOREGROUND'
[Thu Feb 25 12:29:32.092324 2016] [mpm_prefork:notice] [pid 6742] AH00170: caught SIGWINCH, shutting down gracefully
[Thu Feb 25 12:30:17.221041 2016] [core:error] [pid 799] (2)No such file or directory: AH00099: could not create /var/run/httpd-default/httpd.pid
[Thu Feb 25 12:30:17.238833 2016] [core:error] [pid 799] AH00100: httpd-default: could not log pid to file /var/run/httpd-default/httpd.pid
[Thu Feb 25 12:35:39.383811 2016] [mpm_prefork:notice] [pid 4982] AH00163: Apache/2.4.6 (CentOS) PHP/5.4.16 configured -- resuming normal operations
[Thu Feb 25 12:35:39.383845 2016] [core:notice] [pid 4982] AH00094: Command line: '/usr/sbin/httpd-default -f /etc/httpd-default/conf/httpd.conf -D FOREGROUND'
[Thu Feb 25 12:44:43.161934 2016] [mpm_prefork:notice] [pid 4982] AH00170: caught SIGWINCH, shutting down gracefully
[Thu Feb 25 12:45:22.251513 2016] [core:error] [pid 804] (2)No such file or directory: AH00099: could not create /var/run/httpd-default/httpd.pid
[Thu Feb 25 12:45:22.252146 2016] [core:error] [pid 804] AH00100: httpd-default: could not log pid to file /var/run/httpd-default/httpd.pid
[Thu Feb 25 12:54:12.238621 2016] [core:error] [pid 798] (2)No such file or directory: AH00099: could not create /var/run/httpd-default/httpd.pid
[Thu Feb 25 12:54:12.239090 2016] [core:error] [pid 798] AH00100: httpd-default: could not log pid to file /var/run/httpd-default/httpd.pid
[Thu Feb 25 12:56:34.223280 2016] [mpm_prefork:notice] [pid 4971] AH00163: Apache/2.4.6 (CentOS) PHP/5.4.16 configured -- resuming normal operations
[Thu Feb 25 12:56:34.223313 2016] [core:notice] [pid 4971] AH00094: Command line: '/usr/sbin/httpd-default -f /etc/httpd-default/conf/httpd.conf -D FOREGROUND'
[Thu Feb 25 13:09:47.714565 2016] [mpm_prefork:notice] [pid 4971] AH00170: caught SIGWINCH, shutting down gracefully
[Thu Feb 25 13:12:22.223056 2016] [core:error] [pid 801] (2)No such file or directory: AH00099: could not create /var/run/httpd-default/httpd.pid
[Thu Feb 25 13:12:22.248043 2016] [core:error] [pid 801] AH00100: httpd-default: could not log pid to file /var/run/httpd-default/httpd.pid
```

Is this an issue of the httpd cookbook?

**Comment by odoell** [comment]
sorry... i just found #61 

implementing the patch first did not work either. I posted my fix in #61

**Comment by iennae** [comment]
Thanks @odoell, version 0.3.5 of the cookbook should resolve this issue.
